websites/A/Asura Scans/presence.ts

```diff
@@ -1,54 +1,88 @@
-const presence = new Presence({
-		clientId: "864304063804997702",
-	}),
-	browsingTimestamp = Math.floor(Date.now() / 1000);
+const presence = new Presence({ clientId: "864304063804997702" }),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	ASURA_SCANS_LOGO =
+		"https://cdn.rcd.gg/PreMiD/websites/A/Asura%20Scans/assets/logo.png",
+	CHAPTER_PROGRESS_SELECTOR =
+		"body > div:nth-child(4) > div > div > div > div.py-8.-mx-5.md\\:mx-0.flex.flex-col.items-center.justify-center";
 
-presence.on("UpdateData", () => {
+class Comic {
+	title: string;
+	url: string;
+	image: string;
+}
+
+const comic = new Comic();
+
+presence.on("UpdateData", async () => {
 	const { pathname, href } = window.location,
 		presenceData: PresenceData = {
 			startTimestamp: browsingTimestamp,
 			largeImageKey:
 				"https://cdn.rcd.gg/PreMiD/websites/A/Asura%20Scans/assets/logo.png",
-		};
+			type: ActivityType.Watching,
+		},
+		displayPercentage = await presence.getSetting<boolean>("readingPercentage"),
+		privacyMode = await presence.getSetting<boolean>("privacy"),
+		displayChapter = await presence.getSetting<boolean>("chapterNumber"),
+		displayCover = await presence.getSetting<boolean>("showCover"),
+		displayButtons = await presence.getSetting<boolean>("showButtons");
```

Comment:
For better efficiency, combine this using `Promise.all` and destructuring.

Reply:
Will just the Promise.all do the destructing ?